electrum/plugin.py

```diff
@@ -65,41 +65,60 @@ class Plugins(DaemonThread):
     pkgpath = os.path.dirname(plugins.__file__)
 
     @profiler
-    def __init__(self, config: SimpleConfig, gui_name):
+    def __init__(self, config: SimpleConfig = None, gui_name = None, cmd_only: bool = False):
+        self.cmd_only = cmd_only  # type: bool
+        self.internal_plugin_metadata = {}
+        self.external_plugin_metadata = {}
+        self.loaded_command_modules = set()  # type: set[str]
+        if cmd_only:
+            # only import the command modules of plugins
+            self.find_plugins()
+            return
         DaemonThread.__init__(self)
+        self.device_manager = DeviceMgr(config)
         self.name = 'Plugins'  # set name of thread
         self.config = config
         self.hw_wallets = {}
         self.plugins = {}  # type: Dict[str, BasePlugin]
-        self.internal_plugin_metadata = {}
-        self.external_plugin_metadata = {}
         self.gui_name = gui_name
-        self.device_manager = DeviceMgr(config)
-        self.find_internal_plugins()
-        self.find_external_plugins()
+        self.find_plugins()
         self.load_plugins()
         self.add_jobs(self.device_manager.thread_jobs())
         self.start()
 
+    def __getattr__(self, item):
+        # to prevent accessing of a cmd_only instance of this class
+        if self.cmd_only:
+            if item == 'logger':
+                # if something tries to access logger and it is not initialized it gets initialized here
+                Logger.__init__(self)
+                return self.logger
+            raise Exception(f"This instance of Plugins is only for command importing, cannot access {item}")
+
     @property
     def descriptions(self):
         return dict(list(self.internal_plugin_metadata.items()) + list(self.external_plugin_metadata.items()))
 
     def find_internal_plugins(self):
-        """Populates self.internal_plugin_metadata
-        """
+        """Populates self.internal_plugin_metadata"""
         iter_modules = list(pkgutil.iter_modules([self.pkgpath]))
         for loader, name, ispkg in iter_modules:
             # FIXME pyinstaller binaries are packaging each built-in plugin twice:
             #       once as data and once as code. To honor the "no duplicates" rule below,
             #       we exclude the ones packaged as *code*, here:
             if loader.__class__.__qualname__ == "PyiFrozenImporter":
                 continue
-            full_name = f'electrum.plugins.{name}'
+            full_name = f'electrum.plugins.{name}' + ('.commands' if self.cmd_only else '')
             spec = importlib.util.find_spec(full_name)
-            if spec is None:  # pkgutil found it but importlib can't ?!
+            if spec is None:
+                if self.cmd_only:
+                    continue # no commands module in this plugin
```

Comment:
Could you add a command to one of the existing plugins to showcase this functionality?
For example, for the `labels` plugin a force-pull command, similar to the GUI buttons.

See
https://github.com/spesmilo/electrum/blob/e62a2c43c5e1e9769e40ea0fcc2591f5c48166a4/electrum/plugins/labels/qt.py#L51
https://github.com/spesmilo/electrum/blob/e62a2c43c5e1e9769e40ea0fcc2591f5c48166a4/electrum/plugins/labels/labels.py#L180

Reply:
done